pkg/cvo/sync_worker.go

```diff
@@ -246,6 +246,9 @@ func (w *SyncWorker) NotifyAboutManagedResourceActivity(message string) {
 // more resources which are enabled in the current payload. All such capabilities are returned along with any previously
 // existing implicitly enabled capabilities. If no new implicitly enabled capabilities are found, just the previously
 // existing implicitly enabled capabilities are returned.
+//
+// SyncWorker must be locked before syncPayload is called. It yields the lock while retrieving the payload image
+// (potentially long-running IO operation) but acquires it immediately after.
```

Comment:
And this is only partially true until #1005 merges, so I'll remove it from here and include it in the other PR

/lgtm cancel
/hold

Reply:
done